src/detect-engine-analyzer.c

```diff
@@ -924,6 +925,13 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
                 jb_close(js);
                 break;
             }
+            case DETECT_ICMP_ID: {
+                const DetectIcmpIdData *cd = (const DetectIcmpIdData *)smd->ctx;
+                jb_open_object(js, "id");
+                jb_set_uint(js, "number", cd->id);
```

Comment:
Nice job, Alice! We need this little patch here to get this going w your s-v PR:

```
diff --git a/src/detect-engine-analyzer.c b/src/detect-engine-analyzer.c
index a5e8e293b..3d4d8991a 100644
--- a/src/detect-engine-analyzer.c
+++ b/src/detect-engine-analyzer.c
@@ -928,7 +928,7 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
             case DETECT_ICMP_ID: {
                 const DetectIcmpIdData *cd = (const DetectIcmpIdData *)smd->ctx;
                 jb_open_object(js, "id");
-                jb_set_uint(js, "number", cd->id);
+                jb_set_uint(js, "number", SCNtohs(cd->id));
                 jb_close(js);
                 break;
             }
```

This is because we  store ICMP IDs in network byte order. I do not know the protocol well enough to explain why but saw that what we expect in the output is actually host byte order. `SCNtohs` does that conversion for you. :)